### PR TITLE
Ma/token tests

### PR DESCRIPTION
**Pull request by markusahlstrand**
The token endpoint's for auth2 wasn't working for the auth0-spa-js library, so I moved some tests over from the token-service which caused a lot of issues.. Had to patch quite a few things to get all the tests tests to pass. Deployed this to dev to test it out and it seems to be working fine at least for the pkce flow.

### CODE REVIEW

**Comment by danielgent** [comment]
> The token endpoint's for auth2 wasn't working for the auth0-spa-js library

what project is using the `auth0-spa-js library` library? 

What wasn't working?

File: test/routes/tsoa/token.spec.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+      };
+
+      await controller.token({ ctx } as RequestWithContext, tokenParams);
+
+      const cookie = controller.getHeader("set-cookie");
+
+      expect(typeof cookie).toBe("string");
+    });
+
+    it("should throw an error if the vendorId in the state does not match the vendorId of the client", async () => {
```
Comment (danielgent) [inline test/routes/tsoa/token.spec.ts:235]: Cool Markus. These tests explain what wasn't working :+1:

**Comment by markusahlstrand** [comment]
The react-admin thing has a plugin for auth0 that uses the auth0-spa-js. Think we never implemented the token endpoint properly before..

**Comment by danielgent** [comment]
Is `react-admin` used on the auth2 dashboard? or was that the original admin-dashboard template?

**Pull request merged**